src/onnx/parse_loop.cpp

```diff
@@ -54,7 +54,9 @@ struct parse_loop : op_parser<parse_loop>
             auto arg_iters = args.at(0)->eval();
             if(not arg_iters.empty())
             {
-                max_iterations = arg_iters.at<int64_t>();
+                // cap max_iter because loop uses static shapes with max_iter size and huge numbers
+                // here can cause overflow
+                max_iterations = std::min(arg_iters.at<int64_t>(), parser.limit_max_iterations);
```

Comment:
I find it would be better to print message when it is clipping loop iteration to maximum instead of silently clipping it.